pkg/controller/configmap/configmap_controller.go

```diff
@@ -0,0 +1,130 @@
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// Add creates a new ConfigMap Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new ReconcileConfigMap.
+func newReconciler(mgr manager.Manager) *ReconcileConfigMap {
+	client := mgr.GetClient()
+	return &ReconcileConfigMap{
+		client:  client,
+		handler: ca.NewHandler(client),
+	}
+	return &ReconcileConfigMap{}
+}
+
+// add adds a new Controller to mgr with r as the ReconcileConfigMap.
+func add(mgr manager.Manager, r *ReconcileConfigMap) error {
+	if !mktconfig.IsAPIAvailable() {
+		return nil
+	}
+
+	// Create a new controller
+	c, err := controller.New("configmap-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource ConfigMap.
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, getPredicateFunctions())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getPredicateFunctions returns the predicate functions used to identify the configmap
+// that contains Certificate Authority information.
+// True should only be returned when the ConfigMap is updated by the cert-injector-controller.
+func getPredicateFunctions() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			// If the ConfigMap is created we should kick off an event.
+			if e.Meta.GetName() == ca.TrustedCaConfigMapName {
+				return true
+			}
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// If the ConfigMap is updated we should kick off an event.
+			if e.MetaOld.GetName() == ca.TrustedCaConfigMapName {
+				return true
+			}
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+	}
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigMap{}
+
+// ReconcileConfigMap reconciles a ConfigMap object.
+type ReconcileConfigMap struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client  client.Client
+	handler ca.Handler
+}
+
+// Reconcile will restart the marketplace operator if the Certificate Authority ConfigMap is
+// not in sync with the Certificate Authority bundle on disk..
+func (r *ReconcileConfigMap) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Check if the CA ConfigMap is in the same namespace that Marketplace is deployed in.
+	objectInOtherNamespace, err := shared.IsObjectInOtherNamespace(request.Namespace)
```

Comment:
Can we rename this variable to something slightly different? I think using `object` is hard to read. Maybe reference the cert? `certConfigInOtherNamespace`?

Reply:
Sure - I had just followed the convention where this method was used elsewhere.